src/open_inwoner/react/lib/web-component/loader.tsx

```diff
@@ -0,0 +1,186 @@
+    // @ts-expect-error
+    ...(window.IS_DEV ? [performancePlugin] : []),
+  ];
+
+  /**
+   * Find all unique web component names on the current page
+   * @returns a unique array of strings from the founded component names.
+   */
+  private static findWebComponentsOnPage(): WebComponentTagName[] {
+    const selector = Object.keys(WebComponentLoader.registry).join(',');
+    const elements = document.querySelectorAll<HTMLElement>(selector);
+    const foundComponents = [...elements].map((el) => el.tagName.toLowerCase());
+    return Array.from(new Set(foundComponents)) as WebComponentTagName[];
+  }
+
+  /**
+   * Create context objects for all elements of a specific component
+   */
+  private static createContextsForComponent(
+    componentName: string
+  ): WebComponentContext[] {
+    const elements = document.querySelectorAll<HTMLElement>(componentName);
+    return [...elements].map((element) => ({ componentName, element }));
+  }
+
+  private static async runBeforeLoadHooks(context: WebComponentContext) {
+    for (const plugin of WebComponentLoader.pluginRegistry) {
+      try {
+        await plugin.beforeLoad?.(context);
+      } catch (error) {
+        console.error(
+          `[Plugin:${plugin.name}] beforeLoad hook failed for ${context.componentName}:`,
+          error
+        );
+      }
+    }
+  }
+
+  private static async runAfterLoadHooks(context: WebComponentContext) {
+    for (const plugin of WebComponentLoader.pluginRegistry) {
+      try {
+        await plugin.afterLoad?.(context);
+      } catch (error) {
+        console.error(
+          `[Plugin:${plugin.name}] afterLoad hook failed for ${context.componentName}:`,
+          error
+        );
+      }
+    }
+  }
+
+  /**
+   * Error hook runner used to execute the onError plugin function
+   * @param context
+   * @param error The error that occured.
+   */
+  private static async runErrorHooks(
+    context: WebComponentContext,
+    error: Error
+  ) {
+    if (!error) return;
+
+    for (const plugin of WebComponentLoader.pluginRegistry) {
+      try {
+        plugin.onError?.(context, error);
+      } catch (pluginError) {
+        console.error(
+          `[Plugin:${plugin.name}] onError hook failed for ${context.componentName}:`,
+          pluginError
+        );
+      }
+    }
+  }
+
+  private static registerWebComponent<P = {}>(
+    Component: FC<P>,
+    tagName: string,
+    propNames?: (keyof P)[],
+    options: WebComponentRegisterOptions = { shadow: false, i18n: false }
+  ): HTMLElement {
+    // Remove i18n from options before passing to preact-custom-element
+    const { i18n, ...registerOptions } = options;
+
+    // If i18n option is enabled, wrap the component with IntlProvider
+    const ComponentToRegister = i18n ? withIntlWc(Component) : Component;
+
+    return register(ComponentToRegister, tagName, propNames, registerOptions);
+  }
+
+  public static async importWC(name: WebComponentTagName): Promise<void> {
```

Comment:
Generally, I wonder whether short names really help much compared to just reading the noun fully (e.g. optimize for reading, not writing):

```suggestion
  public static async importWebComponent(name: WebComponentTagName): Promise<void> {
```

Reply:
🚾